app/pages/Accounts/useAccountSync.ts

```diff
@@ -73,45 +62,61 @@ export default function useAccountSync(onError: (message: string) => void) {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [account?.status, account?.selfAmounts, account?.incomingAmounts]);
 
-    useEffect(() => {
-        (async () => {
-            if (
-                account?.status !== AccountStatus.Confirmed ||
-                updateLock.isLocked() // If update is already running, we don't need to run again because of new transactions (accountInfo.accountAmount).
-            ) {
-                return;
-            }
-
-            const unlock = await updateLock.acquire();
-
-            const update = dispatch(
-                updateTransactions({
-                    onFirstLoop() {
-                        setUpdateLooped(true);
-                    },
-                    onError: (message) =>
-                        onError(updateTransactionsFailedMessage(message)),
-                })
-            );
-
-            abortUpdateRef.current = update.abort;
-
-            await update;
-
-            unlock();
-            setUpdateLooped(false);
-        })();
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [accountInfo?.accountAmount, account?.status]);
-
     useEffect(
         () => () => {
             abortUpdateRef.current();
         },
         []
     );
 
+    const newTransactionsDependencyArray = [
+        accountInfo?.accountAmount,
+        JSON.stringify(accountInfo?.accountEncryptedAmount.incomingAmounts),
+    ];
     useEffect(() => {
+        if (loadIsDone) {
+            setNewTransactionsFlag(!newTransactionsFlag);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, newTransactionsDependencyArray);
+
+    // Load any new transactions if the account amount changes, as that indicates that a
+    // transaction affected the account.
+    useEffect(() => {
+        if (loadIsDone) {
+            const load = dispatch(
+                loadNewTransactions({
+                    showLoading: true,
+                })
+            );
+
+            return () => {
+                load.abort();
+            };
+        }
+
+        return () => {};
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [newTransactionsFlag]);
+
+    // Re-load transactions entirely from the wallet proxy if:
+    // - the filter is changed
+    // - the status of the account changes, e.g. if it is confirmed,
+    // - the user switches between viewing shielded or unshielded transactions
+    // given that the account info has already been loaded.
+    const loadDependencyArray = [
+        JSON.stringify(account?.transactionFilter),
+        account?.status,
+        accountInfoLoaded,
+        viewingShielded,
+    ];
+    useEffect(() => {
```

Comment:
I was able to make endless load, and duplicate transactions in the list, on a finalizer. Though not consistently, I think it needs to be loading while an accountInfo update triggers a "loadNewTransactions"

Reply:
Seems I can't do this anymore.